include/picongpu/particles/collision/binary/DefaultAlg.hpp

```diff
@@ -0,0 +1,652 @@
+                         * @param gammaComs @f[ \gamma_C @f] Lorentz factor of the COM frame in the lab frame
+                         * @param factorA @f[ \frac{\gamma_C - 1}{v_C^2} @f]
+                         * @param comsVelocity @f[ v_C @f] COM system velocity in the lab frame
+                         */
+                        static DINLINE float3_COLL labToComs(
+                            float3_COLL labMomentum,
+                            float_COLL mass,
+                            float_COLL gamma,
+                            float_COLL gammaComs,
+                            float_COLL factorA,
+                            float3_COLL comsVelocity)
+                        {
+                            float3_COLL labVelocity = labMomentum / gamma / mass;
+                            float_COLL dot = pmacc::math::dot(comsVelocity, labVelocity);
+                            float_COLL factor = (factorA * dot - gammaComs);
+                            factor *= mass * gamma;
+                            float3_COLL diff = factor * comsVelocity;
+
+                            PMACC_DEVICE_ASSERT(std::isfinite((labMomentum + diff)[0]));
+                            PMACC_DEVICE_ASSERT(std::isfinite((labMomentum + diff)[1]));
+                            PMACC_DEVICE_ASSERT(std::isfinite((labMomentum + diff)[2]));
+                            return labMomentum + diff;
+                        }
+
+                        /* Convert momentum from the COM frame into the lab frame.
+                         *
+                         * @param labMomentum momentum in the COM frame
+                         * @param mass particle mass
+                         * @param gamma particle Lorentz factor in the lab frame
+                         * @param gammaComs @f[ \gamma_C @f] Lorentz factor of the COM frame in the lab frame
+                         * @param factorA @f[ \frac{\gamma_C - 1}{v_C^2} @f]
+                         * @param comsVelocity @f[ v_C @f] COM system velocity in the lab frame
+                         */
+                        static DINLINE float3_COLL comsToLab(
+                            float3_COLL comsMomentum,
+                            float_COLL mass,
+                            float_COLL coeff,
+                            float_COLL gammaComs,
+                            float_COLL factorA,
+                            float3_COLL comsVelocity)
+                        {
+                            // (13) in [Perez 2012]
+                            // TODO: try upgrading this to 64bit as well
+                            float_COLL dot = pmacc::math::dot(comsVelocity, comsMomentum);
+                            float_COLL factor = (factorA * dot + coeff * gammaComs);
+                            float3_COLL diff = factor * comsVelocity;
+                            PMACC_DEVICE_ASSERT(std::isfinite((comsMomentum + diff)[0]));
```

Comment:
This should not run on the device because `std::isfinite` is host only. For cuda there is a `isfinite` but this is not available in alpaka.

Reply:
what about isnan and simiilar? std::isfinite was working fine but probably because it's using cuda implementation directly.  I guess the reason why it is not in alpaka is that it wouldn't work on amd?

Reply:
everything which is not `constexpr` e.g https://en.cppreference.com/w/cpp/numeric/math/isnan should not be available on GPU.